src/main/org/deidentifier/arx/DataSource.java

```diff
@@ -1,343 +1,340 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.deidentifier.arx;
+
+import org.deidentifier.arx.io.*;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.sql.SQLException;
+
+/**
+ * This class provides configuration options for importing data from CSV-files, from Excel-files
+ * or via a JDBC connection.
+ *
+ * @author Fabian Prasser
+ * @author Florian Kohlmayer
+ */
+public class DataSource { // NO_UCD
+
+    /**
+     * Creates a CSV data source.
+     *
+     * @param file
+     * @param separator
+     * @param containsHeader
+     * @return
+     */
+    public static DataSource createCSVSource(File file, Charset charset, char separator, boolean containsHeader) {
+        return new DataSource(file, charset, separator, containsHeader);
+    }
+
+    /**
+     * Creates a CSV data source.
+     *
+     * @param file
+     * @param separator
+     * @param containsHeader
+     * @return
+     */
+    public static DataSource createCSVSource(String file, Charset charset, char separator, boolean containsHeader) {
+        return createCSVSource(new File(file), charset, separator, containsHeader);
+    }
+
+    /**
+     * Creates an Excel data source.
+     *
+     * @param file
+     * @param sheetIndex
+     * @param containsHeader
+     * @return
+     */
+    public static DataSource createExcelSource(File file, int sheetIndex, boolean containsHeader) {
+        return new DataSource(file, sheetIndex, containsHeader);
+    }
+
+    /**
+     * Creates an Excel data source.
+     *
+     * @param file
+     * @param sheetIndex
+     * @param containsHeader
+     * @return
+     */
+    public static DataSource createExcelSource(String file, int sheetIndex, boolean containsHeader) {
+        return createExcelSource(new File(file), sheetIndex, containsHeader);
+    }
+
+    /**
+     * Creates a JDBC data source.
+     *
+     * @param url
+     * @param table
+     * @return
+     * @throws SQLException
+     */
+    public static DataSource createJDBCSource(String url, String table) throws SQLException {
+        return new DataSource(url, table);
+    }
+
+    /**
+     * Creates a JDBC data source.
+     *
+     * @param url
+     * @param user
+     * @param password
+     * @param table
+     * @return
+     * @throws SQLException
+     */
+    public static DataSource createJDBCSource(String url, String user, String password, String dbName, String schemaName, String table, int maxRows) throws SQLException {
```

Comment:
It would be good to leave the method where the dbName and schemaName is not specified and add a new method instead where this additional information can be specified.